### PR TITLE
2fa module

### PR DESCRIPTION
**Pull request by saad-abid-crowdbotics**
## Ticket
2FA module

## Type of PR

- [ ] Bugfix
- [x] New feature
- [ ] Minor changes

## Changes introduced

2FA Module integration.

### Verify code 
By clicking "Verify", the code provided is verified against the code sent to phone number or email

### Select verification method
By clicking 'choose verification method', verification method can be selected for authentication

### Verification by email
By clicking the "Email" button, verification code is sent to user email.

### Verification by sms
By clicking the "sms" button, verification code is sent to user phone number.

### Verification by 2FA
By clicking the "2FA" button, authentication through google Authenticator generated token can be achieved.

### Set up 2FA on same device
By clicking "set up on same device", Google Authenticator account sets up automatically

### QR code 
By scanning the QR code, Google Authenticator account sets up automatically.

### Resend code
By clicking "Resend", the code is resent to Phone number or email

image
image
image
image

### CODE REVIEW

File: modules/two-factor-authentication/backend/modules/two-factor-authentication/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Generated by Django 2.2.23 on 2021-06-19 20:54
```
Comment (danielsousaio) [inline modules/two-factor-authentication/backend/modules/two-factor-authentication/migrations/0001_initial.py:1]: As the very last step of this review process, lets squash the migrations into a single one.

File: modules/two-factor-authentication/backend/modules/two-factor-authentication/viewsets.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
+import pyotp
+from rest_framework import status
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.viewsets import ModelViewSet
+
+from demo import settings
+from demo.settings import ACCOUNT_SID, AUTH_TOKEN, SENDGRID_API_KEY
```
Comment (danielsousaio) [inline modules/two-factor-authentication/backend/modules/two-factor-authentication/viewsets.py:9]: This should use `from django.conf import settings` instead

File: modules/two-factor-authentication/backend/modules/two-factor-authentication/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Two Factor Authentication
```
Comment (danielsousaio) [inline modules/two-factor-authentication/backend/modules/two-factor-authentication/README.md:1]: Missing required dependencies installation and configuration details.

**Review by danielsousaio — CHANGES_REQUESTED** [review]

File: modules/two-factor-authentication/backend/modules/two-factor-authentication/README.md (hunk 3 above)
Comment (saad-abid-crowdbotics) [reply modules/two-factor-authentication/backend/modules/two-factor-authentication/README.md:1]: Resolved

**Review by saad-abid-crowdbotics — COMMENTED** [review]
Resolved requested changes.

**Comment by shahraizali** [comment]
@saad-abid-crowdbotics Update backend structure

**Pull request closed without merging**